src/GraphQL/Extensions/GraphQLBuilderExtensions.cs

```diff
@@ -652,6 +652,33 @@ public static IGraphQLBuilder AddDocumentListener<TDocumentListener>(this IGraph
     }
     #endregion
 
+    #region - AddResolveFieldContextAccessor -
+    /// <summary>
+    /// Enables the <see cref="IResolveFieldContextAccessor"/> feature, which allows user code to retrieve
+    /// the current <see cref="IResolveFieldContext"/> during field resolution. This is similar to
+    /// <c>IHttpContextAccessor</c> in ASP.NET Core. Registers <see cref="IResolveFieldContextAccessor"/>
+    /// as a singleton and configures the schema to populate the accessor during field execution.
+    /// </summary>
+    public static IGraphQLBuilder AddResolveFieldContextAccessor(this IGraphQLBuilder builder)
+    {
+        // Register the accessor as a singleton
+        builder.Services.TryRegister<IResolveFieldContextAccessor>(ResolveFieldContextAccessor.Instance);
+
+        // Configure the schema to use the accessor
+        builder.ConfigureSchema<ConfigureResolveFieldContextAccessor>();
```

Comment:
The design allows for `AddResolveFieldContextAccessor` to be called any number of times.